File: src/main/java/com/jss/osiris/modules/quotation/service/DomiciliationFeeService.java
package com.jss.osiris.modules.quotation.service;

import com.jss.osiris.modules.quotation.model.DomiciliationFee;

public interface DomiciliationFeeService {
    public DomiciliationFee getDomiciliationFee(Integer id);

    public void deleteDomiciliationFee(DomiciliationFee domiciliationFee);

    public DomiciliationFee addDomiciliationFee(DomiciliationFee domiciliationFee);
}

File: client/src/app/modules/quotation/services/domiciliation.fee.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DomiciliationFee } from '../model/DomiciliationFee';

@Injectable({
  providedIn: 'root'
})
export class DomiciliationFeeService extends AppRestService<DomiciliationFee> {
  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  addDomiciliationFee(newFee: DomiciliationFee) {
    return this.addOrUpdate(new HttpParams(), "domiciliation/fee/add", newFee);
  }

  deleteDomiciliationFee(fee: DomiciliationFee) {
    return this.get(new HttpParams().set("domiciliationFeeId", fee.id), "domiciliation/fee/delete");
  }

}
